### PR TITLE
KAFKA-12744: Breaking change dependency upgrade: "argparse4j" 0.7.0 -->> 0.9.0

### PR DESCRIPTION
**Pull request by dejan2609**
Corresponding JIRA ticket: https://issues.apache.org/jira/browse/KAFKA-12744 

Migration guide: https://argparse4j.github.io/migration.html#to-0-8-0

Release notes:
 * https://github.com/argparse4j/argparse4j/releases/tag/argparse4j-0.8.0
 * https://github.com/argparse4j/argparse4j/releases/tag/argparse4j-0.8.1
 * https://github.com/argparse4j/argparse4j/releases/tag/argparse4j-0.9.0

Related to this PR: #10094 and this commit: https://github.com/apache/kafka/commit/690f72dd69d31589655c84d3cc1a6eec006bcab5

@cmccabe, @hachikuji, @mumrah and @soarez: it would be nice if you could spare some time and review this (especially these two classes):

```
shell/src/main/java/org/apache/kafka/shell/Commands.java
shell/src/main/java/org/apache/kafka/shell/ManCommandHandler.java
```

FYI also @ijuma

### CODE REVIEW

File: shell/src/main/java/org/apache/kafka/shell/Commands.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ void completeNext(MetadataNodeManager nodeManager,
      * @param addShellCommands  True if we should include the shell-only commands.
      */
     public Commands(boolean addShellCommands) {
-        this.parser = ArgumentParsers.newArgumentParser("", false);
+        this.parser = ArgumentParsers.newFor("shell-commands").build();
```
Comment (dejan2609) [inline shell/src/main/java/org/apache/kafka/shell/Commands.java:98]: Note: I think need a review for this.

File: shell/src/main/java/org/apache/kafka/shell/ManCommandHandler.java [hunk 2]
```diff
@@ -87,7 +87,7 @@ public void run(Optional<InteractiveShell> shell,
             writer.println("man: unknown command " + cmd +
                 ". Type help to get a list of commands.");
         } else {
-            ArgumentParser parser = ArgumentParsers.newArgumentParser(type.name(), false);
+            ArgumentParser parser = ArgumentParsers.newFor(type.name()).build();
```
Comment (dejan2609) [inline shell/src/main/java/org/apache/kafka/shell/ManCommandHandler.java:90]: This also needs a review.

**Comment by ijuma** [comment]
Thanks for the PR. Given that testing for CLI tools is usually not as good as it should be, have you tried running the commands yourself after these changes?

**Comment by dejan2609** [comment]
@ijuma To be honest: no, I did not test commands directly (thing is that **_Kafka CLI tools_** are a new concept for me). 
Just to be sure: I presume you are referring to these CLI tools:
- https://docs.confluent.io/platform/current/tutorials/examples/clients/docs/kafka-commands.html
- https://docs.cloudera.com/documentation/kafka/latest/topics/kafka_command_line.html
- https://docs.cloudera.com/runtime/7.2.9/kafka-managing/topics/kafka-manage-cli-overview.html

Feel free to drop me some related resource / example / github repository (if you know one) so I can test/run commands on my end.

**Comment by dejan2609** [comment]
Update / mental note: basically I will have to create a binary out of this PR, start Zookeeper, run Kafka Broker and so on:  https://dzone.com/articles/apache-kafka-basic-setup-and-usage-with-command-li

After that I will test as much as possible:
- https://gist.github.com/ursuad/e5b8542024a15e4db601f34906b30bb5
- https://www.tutorialspoint.com/apache_kafka/apache_kafka_basic_operations.htm 
- https://kontext.tech/column/streaming-analytics/476/kafka-windows-cli-commands 
- https://medium.com/@TimvanBaarsen/apache-kafka-cli-commands-cheat-sheet-a6f06eac01b 
- https://docs.cloudera.com/documentation/enterprise/6/6.3/topics/kafka_admin_cli.html
- https://myknowpega.com/2020/06/30/kafka-2-set-up-and-cli-tutorial/

**Comment by dejan2609** [comment]
Changing status to draft (until test CLI, as mentioned in a comments above),

**Comment by mimaison** [comment]
We can't upgrade to argparse 0.9.0 now as it requires Java 9. Closing this PR for now, we can revisit it once Kafka drops support for Java 8.

**Comment by dejan2609** [comment]
Hi @mimaison ! I will return to this once Kafka drops support for Java 8.

Reference JIRA ticket:  
**_KIP-750: Drop support for Java 8 in Kafka 4.0 (deprecate in 3.0)_** https://issues.apache.org/jira/browse/KAFKA-12894

**Pull request closed without merging**